osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs

```diff
@@ -62,4 +81,78 @@ public enum BeatmapSearchCategory
         [Description("My Maps")]
         Mine,
     }
+
+    public enum BeatmapSearchGenre
+    {
+        Any = 0,
+        Unspecified = 1,
+
+        [Description("Video Game")]
+        VideoGame = 2,
+        Anime = 3,
+        Rock = 4,
+        Pop = 5,
+        Other = 6,
+        Novelty = 7,
+
+        [Description("Hip Hop")]
+        HipHop = 9,
+        Electronic = 10
+    }
+
+    [HasOrderedElements]
+    public enum BeatmapSearchLanguage
+    {
+        [Order(0)]
+        Any,
+
+        [Order(11)]
+        Other,
+
+        [Order(1)]
+        English,
+
+        [Order(6)]
+        Japanese,
+
+        [Order(2)]
+        Chinese,
+
+        [Order(10)]
+        Instrumental,
+
+        [Order(7)]
+        Korean,
+
+        [Order(3)]
+        French,
+
+        [Order(4)]
+        German,
+
+        [Order(9)]
+        Swedish,
+
+        [Order(8)]
+        Spanish,
+
+        [Order(5)]
+        Italian
+    }
+
+    [AttributeUsage(AttributeTargets.Field)]
+    public class OrderAttribute : Attribute
+    {
+        public readonly int Order;
+
+        public OrderAttribute(int order)
+        {
+            Order = order;
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Enum)]
+    public class HasOrderedElementsAttribute : Attribute
+    {
+    }
```

Comment:
Does not belong here & might be useful elsewhere later. I'd move to somewhere common - `osu.Game.Utils` maybe?